File: circle.py
import math

from i_figure import IFigure

class Circle(IFigure):
    def __init__(self, radius):
        assert radius > 0, "Radius must be greater than 0"
        self.radius = radius

    @property
    def square(self):
        return math.pi * self.radius**2

File: CsharpSquare/Circle.cs
public class Circle : IShape
{
    public double Radius { get; }

    public Circle(double radius)
    {
        ArgumentOutOfRangeException.ThrowIfNegativeOrZero(radius, nameof(radius));
        Radius = radius;
    }

    public double Square()
    {
        return Math.PI * Math.Pow(Radius, 2);
    }
}